test/FastTests/Client/Indexing/JavaScriptIndexTests.cs

```diff
@@ -212,6 +303,108 @@ public override IndexDefinition CreateIndexDefinition()
             }
         }
 
+        private class FanoutByNumbers : AbstractIndexCreationTask
+        {
+            public override IndexDefinition CreateIndexDefinition()
+            {
+                return new IndexDefinition
+                {
+                    Name = "FanoutByNumbers",
+                    Maps = new HashSet<string>
+                    {
+                        @"map('Fanouts', function (f){ 
+ var result = [];
+for(var i = 0; i < f.Numbers.length; i++)
+{
+    result.push({
+        Foo: f.Foo,
+        Sum: f.Numbers[i]
+    });
+}
+return result;
+})",
+                    },
+                    Type = IndexType.JavaScriptMap,
+                    LockMode = IndexLockMode.Unlock,
+                    Priority = IndexPriority.Normal,
+                    Configuration = new IndexConfiguration()
+                };
+            }
+
+            internal class Result
+            {
+                public string Foo { get; set; }
+                public int Sum { get; set; } 
+            }
+        }
+
+        private class FanoutByNumbersWithReduce : AbstractIndexCreationTask
+        {
+            public override IndexDefinition CreateIndexDefinition()
+            {
+                return new IndexDefinition
+                {
+                    Name = "FanoutByNumbersWithReduce",
+                    Maps = new HashSet<string>
+                    {
+                        @"map('Fanouts', function (f){ 
+ var result = [];
+for(var i = 0; i < f.Numbers.length; i++)
+{
+    result.push({
+        Foo: f.Foo,
+        Sum: f.Numbers[i]
+    });
+}
+return result;
+})",
+                    },
+                    Reduce =
+                    @"
+groupBy( f => f.Foo )
+ .aggregate( g => ({
+     Foo: g.key,
+     Sum: g.values.reduce((total, val) => val.Sum + total,0)
+ }))",//TODO: do actual reduction here
```

Comment:
why todo, aren't you actually reducing here?

Reply:
Forgot to remove the comment